src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -614,25 +673,17 @@ private int WriteCommonFields(Span<byte> buffer, TarEntryType actualEntryType)
 
             int checksum = 0;
 
-            if (_mode > 0)
-            {
-                checksum += FormatNumeric(_mode, buffer.Slice(FieldLocations.Mode, FieldLengths.Mode));
-            }
+            int mode = Math.Max(0, _mode);
+            checksum += FormatNumeric(mode, buffer.Slice(FieldLocations.Mode, FieldLengths.Mode));
 
-            if (_uid > 0)
-            {
-                checksum += FormatNumeric(_uid, buffer.Slice(FieldLocations.Uid, FieldLengths.Uid));
-            }
+            int uid = Math.Max(0, _uid);
+            checksum += FormatNumeric(uid, buffer.Slice(FieldLocations.Uid, FieldLengths.Uid));
 
-            if (_gid > 0)
-            {
-                checksum += FormatNumeric(_gid, buffer.Slice(FieldLocations.Gid, FieldLengths.Gid));
-            }
+            int gid = Math.Max(0, _gid);
+            checksum += FormatNumeric(gid, buffer.Slice(FieldLocations.Gid, FieldLengths.Gid));
 
-            if (_size >= 0)
-            {
-                checksum += FormatNumeric(_size, buffer.Slice(FieldLocations.Size, FieldLengths.Size));
-            }
+            long size = Math.Max(0, _size);
+            checksum += FormatNumeric(size, buffer.Slice(FieldLocations.Size, FieldLengths.Size));
```

Comment:
@tmds I addressed this for all 4 fields.

Reply:
This applied https://github.com/dotnet/runtime/pull/114940#discussion_r2055500705.

I made a different suggestion in the follow up comment https://github.com/dotnet/runtime/pull/114940#discussion_r2055529535.

The latter is to only update gid and uid as in:
```
if (_uid >= 0)
{
    checksum += FormatNumeric(_uid, buffer.Slice(FieldLocations.Uid, FieldLengths.Uid));
}
```

and leave mode and size unchanged.

Reply:
Ah sorry. Yes, you're right. I'll bring back the if branches.

What about mode? I know it would be _weird_ to see all zeros in the mode field, but if we see some tar in the wild with such value, we should write it as 0x30 chars too.

Reply:
Yes, let's also change `_mode` to be `>= 0`.

Reply:
Fixed